cmd/charm/charmcmd/cmd_test.go

```diff
@@ -40,36 +45,45 @@ import (
 // its context directory set to dir. It returns the output of the command
 // and its exit code.
 func run(dir string, args ...string) (stdout, stderr string, exitCode int) {
-	return runWithInput(dir, "", args...)
-}
-
-// runWithInput runs a charm plugin subcommand with the given arguments,
-// its context directory set to dir, and with standard input set to in.
-// It returns the output of the command and its exit code.
-func runWithInput(dir string, in string, args ...string) (stdout, stderr string, exitCode int) {
 	// Remove the warning writer usually registered by cmd.Log.Start, so that
 	// it is possible to run multiple commands in the same test.
 	// We are not interested in possible errors here.
 	defer loggo.RemoveWriter("warning")
 	var stdoutBuf, stderrBuf bytes.Buffer
 	ctxt := &cmd.Context{
 		Dir:    dir,
-		Stdin:  strings.NewReader(in),
+		Stdin:  nil,
 		Stdout: &stdoutBuf,
 		Stderr: &stderrBuf,
 	}
 	exitCode = cmd.Main(charmcmd.New(), ctxt, args)
 	return stdoutBuf.String(), stderrBuf.String(), exitCode
 }
 
+type homeSuite struct{}
```

Comment:
Let's please not call this "homeSuite" as that sounds like it's got a suite of tests, which it doesn't (it was always an aberration that the juju helper "suites" were named that way). Maybe:

	// setFakeHome sets up environment variables
	// to point to a new temporary home directory,
	// and installs cleanup functions to restore the
	// changed global state.
	func setFakeHome(c *qt.C) {
		oldHome := utils.Home()
		c.AddCleanup(func() {
			utils.SetHome(oldHome)
		})
		utils.SetHome(c.Mkdir())
		c.Setenv("HOME", utils.Home())
		err := os.MkdirAll(jujutesting.JujuXDGDataHomePath(), 0755)
		c.Assert(err, qt.Equals, nil)
		c.AddCleanup(func() {
			osenv.SetJujuXDGDataHome("")
		})
		osenv.SetJujuXDGDataHome(jujutesting.JujuXDGDataHomePath())
	}

?